apps/emqx_authz/test/emqx_authz_SUITE.erl

```diff
@@ -65,7 +65,9 @@ set_special_configs(_App) ->
 
 -define(SOURCE1, #{<<"type">> => <<"http">>,
                    <<"enable">> => true,
-                   <<"url">> => <<"https://fake.com:443/">>,
+                   <<"base_url">> => <<"https://example.com:443/">>,
+                   <<"path">> => <<"a/b">>,
+                   <<"query">> => <<"c=d">>,
```

Comment:
This changes the configuration file.

As a user of emqx, I'd prefer

```
{url = "http://example.com:443/a/b?c=d"}
```

To

```
{ base_url = "http://example.com:443/",
  path = "a/b"
  query = "c=d"
}

```

Reply:
In practice, users will have smth like 
```
{url = "http://example.com:443/${username}/b?c=${topic}"} 
```
And this is not a valid URL.

To make this variant working we would have to chose
* Either to write quite a lot of code for validation and parsing "templated urls".
* Rely on some not quite universal assumptions about such urls, which can lead to strange behaviours in some cases.

Reply:
`query` encode has solved in #6558 
and the other parts may need encoding too.
Maybe we can put the encode funs in [emqx_http_lib](https://github.com/emqx/emqx_http_lib) based on `uri_string` and `http_uri`.